tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +66,699 @@ public function test_wp_enqueue_script() {
+		$expected  = "<script type='text/javascript' src='/wp-admin/load-scripts.php?c=0&amp;load%5Bchunk_0%5D=one-concat-dep,two-concat-dep,three-concat-dep&amp;ver={$ver}'></script>\n";
+		$expected .= "<script type='text/javascript' src='/main-script.js' id='main-defer-script-js' defer></script>\n";
+
+		$this->assertSame( $expected, $print_scripts );
+	}
+
+	/**
+	 * Test script concatenation with `async` main script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_async_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value           = $concatenate_scripts;
+		$concatenate_scripts = true;
+
+		$wp_scripts->do_concat    = true;
+		$wp_scripts->default_dirs = array( $this->default_scripts_dir );
+
+		wp_enqueue_script( 'one-concat-dep-1', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'two-concat-dep-1', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'three-concat-dep-1', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'main-async-script-1', '/main-script.js', array(), null, array( 'strategy' => 'async' ) );
+
+		wp_print_scripts();
+		$print_scripts = get_echo( '_print_scripts' );
+
+		// reset global before asserting.
+		$concatenate_scripts = $old_value;
+
+		$ver       = get_bloginfo( 'version' );
+		$expected  = "<script type='text/javascript' src='/wp-admin/load-scripts.php?c=0&amp;load%5Bchunk_0%5D=one-concat-dep-1,two-concat-dep-1,three-concat-dep-1&amp;ver={$ver}'></script>\n";
+		$expected .= "<script type='text/javascript' src='/main-script.js' id='main-async-script-1-js' async></script>\n";
+
+		$this->assertSame( $expected, $print_scripts );
+	}
+
+	/**
+	 * Test script concatenation with blocking scripts before and after a `defer` script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_blocking_script_before_and_after_script_with_defer_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value           = $concatenate_scripts;
+		$concatenate_scripts = true;
+
+		$wp_scripts->do_concat    = true;
+		$wp_scripts->default_dirs = array( $this->default_scripts_dir );
+
+		wp_enqueue_script( 'one-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'two-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'three-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'deferred-script-2', '/main-script.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'four-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'five-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'six-concat-dep-2', $this->default_scripts_dir . 'script.js' );
+
+		wp_print_scripts();
+		$print_scripts = get_echo( '_print_scripts' );
+
+		// reset global before asserting.
+		$concatenate_scripts = $old_value;
+
+		$ver       = get_bloginfo( 'version' );
```

Comment:
While `get_bloginfo( 'version' )` is used in other tests in this file, a function call isn't needed. We can just use `$wp_version` directly by adding it to the globals on the first line of this test method and using it in the `$expected` value.

Reply:
Done in b50f76531a